python/publish/publisher.py

```diff
@@ -305,6 +307,15 @@ def publish_json(self, data: PublishData):
         # provide a reduced version to Github actions
         self._gha.set_output('json', json.dumps(data.reduced(), ensure_ascii=False))
 
+    def publish_summary(self, data: PublishData):
+        if self._settings.job_summary and  os.getenv("GITHUB_STEP_SUMMARY"):
```

Comment:
Can we add the path to the step summary file to settings, populated from that env var? Please have a check there that if `job_summary` is true, that env var has to be set. This way we only have to check for `self._settings.job_summary` as the path has to be valid then.

Then this file stays free from env vars.

Reply:
That's another implementation choice, I didn't want to expose the github technical aspect to users.
Thé goal of checking the environment variable was more to comply with the linter error reported in the first version of m'y commit

Reply:
> I didn't want to expose the github technical aspect to users.

It won't be exposed to the user. It should become part of `Settings` but not input variables, if that makes sense.

Reply:
Ok, I see your point. would a property do?
Also, as the default is true, there would be a huge load of tests to rewrite, so I'll keep anyway a check on the existence of the file or something to avoid pollution of the tests. Would that be ok?

Reply:
We can make the `job_summary` false throughout the tests (e.g. in [create_settings](https://github.com/EnricoMi/publish-unit-test-result-action/blob/master/python/test/test_publisher.py#L46)) unless we test the job summary itself, or have `job_summary_file` point to `/dev/null` (which might break tests on Windows).

Where do you see huge test rewrites?

I think the existence of the file should not be required as appending to the path would create it. I would insist on it existing. If writing fails, the exception tells us. The path just has to be an non-empty string, that is good enough.

Reply:
I ran into issues with `python/test/test_action_script.py` when I changed the parsing from default False to default True, and I panicked 😅 
But in fact I misread something and the creation of settings object is not the same for all test classes, so it's ok.